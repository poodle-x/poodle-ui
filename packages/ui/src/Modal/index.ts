import React from "react";
import { Modal, ModalProps } from "./Modal";
import { StandardThemeConfig } from "../theme";

export * from "./Modal";

export default Modal;

export type ModalStyleKeys =
	| "root"
	| "overlay"
	| "inner"
	| "container"
	| "modal";

export interface ModalThemeConfig
	extends StandardThemeConfig<ModalStyleKeys, ModalProps> {}

export interface ModalContextValue {
	autoIdLabelledby?: string;
	autoIdDescribedby?: string;
}

export const ModalContext = React.createContext<ModalContextValue>({});
