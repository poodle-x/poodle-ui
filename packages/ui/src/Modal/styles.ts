import { getSpacing } from "@poodle/system";
import { createCSSSystemStandard } from "../theme";
import { CSSObjectSystem, CSSSystem } from "../styled";
import { ModalProps } from "./Modal";
import safeThemeUnitValue from "../utils/safeThemeValue";
import { ModalStyleKeys } from "./index";

export function Root(props: ModalProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard<ModalStyleKeys>({
		key: "root",
		config: theme?.Modal,
		props,
		base: [
			{
				position: "fixed",
				top: 0,
				left: 0,
				width: "100%",
				height: "100%",
				zIndex: "modal",
			},
		],
	});
}

export function Overlay(props: ModalProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard<ModalStyleKeys>({
		key: "overlay",
		config: theme?.Modal,
		props,
		base: [
			{
				position: "fixed",
				top: 0,
				left: 0,
				width: "100%",
				height: "100%",
				zIndex: -1,
				backgroundColor: "rgba(0, 0, 0, 0.25)",
			},
		],
	});
}

export function Inner(props: ModalProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard({
		key: "inner",
		config: theme?.Modal,
		props,
		base: [
			{
				display: "flex",
				alignItems: "center",
				justifyContent: "center",
				height: "100%",
				outline: "none",
			},
		],
	});
}

export function Container(props: ModalProps): CSSSystem {
	const { theme } = props;

	const topBottomMargin = getSpacing("scale-20")({ theme });

	return createCSSSystemStandard<ModalStyleKeys>({
		key: "container",
		config: theme?.Modal,
		props,
		base: [
			{
				maxHeight: `calc(100% - ${`${safeThemeUnitValue(topBottomMargin)}`})`,
				mx: "scale-4",
				my: "scale-10",
				width: "100%",
				maxWidth: "600px",
			},
		],
	});
}

export function Modal(props: ModalProps): CSSSystem {
	const { theme, colorStyle } = props;

	let colorStyles: CSSObjectSystem = {};

	switch (colorStyle) {
		case "primary": {
			colorStyles = {
				borderTop: "4px solid",
				borderColor: "primary.base",
			};
			break;
		}
		case "negative": {
			colorStyles = {
				borderTop: "4px solid",
				borderColor: "negative.base",
			};
			break;
		}
		case "positive": {
			colorStyles = {
				borderTop: "4px solid",
				borderColor: "positive.base",
			};
			break;
		}
		case "warn": {
			colorStyles = {
				borderTop: "4px solid",
				borderColor: "warn.base",
			};
			break;
		}
	}

	return createCSSSystemStandard({
		key: "modal",
		config: theme?.Modal,
		props,
		base: [
			{
				width: "100%",
				maxHeight: "100%",
				borderRadius: "4px",
				backgroundColor: "modal.base",
				boxShadow: "lv1",
			},
			colorStyles,
		],
	});
}
