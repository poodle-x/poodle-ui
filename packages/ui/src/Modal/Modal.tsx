import React from "react";
import FocusTrap from "focus-trap-react";
import { Options as FocusTrapOptions } from "focus-trap";
import Portal, { PortalProps } from "../Portal";
import Box, { BoxProps } from "../Box";
import { StandardComponentProps, ThemeConfig } from "../theme";
import useDefaultProps from "../utils/useDefaultProps";
import { useClassNames } from "../styled";
import * as styles from "./styles";
import useAutoId from "../hooks/useAutoId";
import { ModalContext } from ".";

export interface LocalModalProps {
	children?: React.ReactNode;
	/**
	 * Open the modal when **`true`**.
	 */
	isOpen?: boolean;
	/**
	 * Callback called when the user tries to close the modal.
	 *
	 * It can be caused by click on outside the modal or press escape key.
	 */
	onRequestClose?: (cause: "outside" | "esc") => void;
	/**
	 * Custom props for inner element.
	 */
	innerProps?: BoxProps;
	/**
	 * Custom props for container element.
	 */
	containerProps?: BoxProps;
	/**
	 * Custom props for overlay element.
	 */
	overlayProps?: BoxProps;
	/**
	 * Custom props for root element.
	 */
	rootProps?: BoxProps;
	/**
	 * Custom props for modal element.
	 */
	modalProps?: BoxProps;
	/**
	 * Turn off focus trap.
	 */
	disableFocusTrap?: boolean;
	/**
	 * Custom props for **`<Portal />`**.
	 */
	portalProps?: Omit<PortalProps, "mount">;
	/**
	 * Add autoId
	 */
	withAutoId?: boolean;
	/**
	 * Pause or unpause focus trap
	 */
	pausedFocusTrap?: boolean;
	/**
	 * The elements will be used as the boundaries for the focus-trap,
	 * instead of the child.
	 */
	containerElementsFocusTrap?: Array<HTMLElement>;
	/**
	 * Pass any of the options available in focus-trap's createOptions.
	 */
	focusTrapOptions?: FocusTrapOptions;
}

export interface ModalProps
	extends BoxProps,
		StandardComponentProps,
		LocalModalProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.Modal?.defaultProps;
}

export const Modal: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<ModalProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, ModalProps>((_props, ref) => {
	const props = useDefaultProps<ModalProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const {
		children,
		className,
		onRequestClose,
		innerProps,
		containerProps,
		modalProps,
		overlayProps,
		portalProps,
		isOpen,
		disableFocusTrap,
		withAutoId,
		pausedFocusTrap,
		containerElementsFocusTrap,
		focusTrapOptions,
		...otherProps
	} = props;

	const autoIdLabelledby = useAutoId(undefined, {
		base: "poodle-modal-label",
	});
	const autoIdDescribedby = useAutoId(undefined, {
		base: "poodle-modal-describe",
	});

	const mouseDownTarget = React.useRef<HTMLElement | null>(null);

	function handlerOverlayClick(e: React.MouseEvent<HTMLElement>) {
		if (e.target !== e.currentTarget) {
			return;
		}

		if (mouseDownTarget.current !== e.target) {
			return;
		}

		mouseDownTarget.current = null;

		if (onRequestClose) {
			onRequestClose("outside");
		}
	}

	function handlerOverlayMouseDown(e: React.MouseEvent<HTMLElement>) {
		mouseDownTarget.current = e.target as HTMLElement;
	}

	function handleKeyDown(e: React.KeyboardEvent<HTMLElement>) {
		if (e.key !== "Escape") {
			return;
		}

		e.stopPropagation();

		if (onRequestClose) {
			onRequestClose("esc");
		}
	}

	const classes = useClassNames({
		props,
		lists: {
			root: {
				classNames: ["poodle-modal", styles.Root, className],
			},
			overlay: {
				classNames: [
					"poodle-modal__overlay",
					styles.Overlay,
					overlayProps?.className,
				],
			},
			inner: {
				classNames: [
					"poodle-modal__inner",
					styles.Inner,
					innerProps?.className,
				],
			},
			container: {
				classNames: [
					"poodle-modal__container",
					styles.Container,
					containerProps?.className,
				],
			},
			modal: {
				classNames: [
					"poodle-modal__modal",
					styles.Modal,
					modalProps?.className,
				],
			},
		},
	});

	const renderChild = (
		<Box onKeyDown={handleKeyDown} {...otherProps} className={classes.root}>
			<Box aria-hidden={true} {...overlayProps} className={classes.overlay} />
			<Box
				onMouseDown={handlerOverlayMouseDown}
				onClick={handlerOverlayClick}
				{...innerProps}
				className={classes.inner}
			>
				<Box tabIndex={-1} {...containerProps} className={classes.container}>
					<Box
						role="dialog"
						aria-modal={true}
						aria-labelledby={withAutoId ? autoIdLabelledby : undefined}
						aria-describedby={withAutoId ? autoIdDescribedby : undefined}
						{...modalProps}
						className={classes.modal}
					>
						{children}
					</Box>
				</Box>
			</Box>
		</Box>
	);

	return (
		<ModalContext.Provider
			value={{
				autoIdDescribedby: withAutoId ? autoIdDescribedby : undefined,
				autoIdLabelledby: withAutoId ? autoIdLabelledby : undefined,
			}}
		>
			<Portal {...portalProps} ref={ref} mount={isOpen}>
				<FocusTrap
					containerElements={containerElementsFocusTrap}
					paused={pausedFocusTrap}
					active={!disableFocusTrap}
					focusTrapOptions={{
						allowOutsideClick: true,
						...focusTrapOptions,
					}}
				>
					{renderChild}
				</FocusTrap>
			</Portal>
		</ModalContext.Provider>
	);
});

Modal.displayName = "PoodleModal";
