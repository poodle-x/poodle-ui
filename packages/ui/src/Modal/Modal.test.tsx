import { axe } from "jest-axe";
import React from "react";
import Box from "../Box";
import { Button } from "../Button";
import { ModalBody } from "../ModalBody";
import { ModalFooter } from "../ModalFooter";
import { ModalHeader } from "../ModalHeader";
import { ThemeConfig } from "../theme";
import { fireEvent, render, userEvent } from "../utils/test";
import Modal from "./";

describe("<Modal />", () => {
	it("should render Modal correctly", () => {
		const { getByTestId } = render(
			<Modal data-testid="test" isOpen={true}>
				<button>Modal</button>
			</Modal>
		);
		expect(getByTestId("test")).toMatchSnapshot();
	});

	it("should render Modal withAutoId correctly", () => {
		const { getByTestId } = render(
			<Modal data-testid="test" isOpen={true} withAutoId={true}>
				<button>Modal</button>
			</Modal>
		);
		expect(getByTestId("test")).toMatchSnapshot();
	});

	it("should get callback request close with escape key", () => {
		const mockCallback = jest.fn();
		const mockInnerKeyDown = jest.fn();

		const { getByRole } = render(
			<Modal
				data-testid="test"
				onRequestClose={mockCallback}
				isOpen={true}
				withAutoId={true}
				innerProps={{
					onKeyDown: mockInnerKeyDown,
				}}
			>
				<button>Modal</button>
			</Modal>
		);
		const dialog = getByRole("dialog");

		fireEvent.keyDown(dialog, { key: "Escape", code: "Escape" });

		expect(mockCallback).toHaveBeenCalledWith("esc");
		expect(mockInnerKeyDown).toHaveBeenCalled();
	});

	it("should get call request close with click on modal backdrop", async () => {
		const mockCallback = jest.fn();
		const mockInnerClick = jest.fn();
		const user = userEvent.setup();
		const { getByTestId } = render(
			<Modal
				data-testid="1"
				innerProps={{
					"data-testid": "test",
					onClick: mockInnerClick,
				}}
				onRequestClose={mockCallback}
				isOpen={true}
				withAutoId={true}
			>
				<button>Modal</button>
			</Modal>
		);
		const inner = getByTestId("test");

		await user.click(inner);

		expect(mockCallback).toHaveBeenCalledWith("outside");
		expect(mockInnerClick).toHaveBeenCalled();
	});

	it("ref should work", () => {
		const ref = React.createRef<HTMLDivElement>();

		render(
			<Modal isOpen={true} ref={ref}>
				<button>Modal</button>
			</Modal>
		);
		expect(ref.current).toMatchSnapshot();
	});

	it("should pass axe", async () => {
		const { getByRole } = render(
			<Modal isOpen={true} withAutoId={true}>
				<ModalHeader withClose={true}>Modal title</ModalHeader>
				<ModalBody>
					The selected document <Box as="strong">“price.docx”</Box> will be
					sent.
				</ModalBody>
				<ModalFooter>
					<Button variant="fill">Send</Button>
					<Button ml="scale-2">Cancel</Button>
				</ModalFooter>
			</Modal>
		);
		const results = await axe(getByRole("dialog"));
		expect(results).toHaveNoViolations();
	});

	it("should render Box props correctly", () => {
		const { getByTestId } = render(
			<Modal
				isOpen={true}
				data-testid="test"
				id="id"
				mt="20px"
				overlayProps={{ ml: 20 }}
				innerProps={{ pl: 20 }}
				containerProps={{ pl: 20 }}
				modalProps={{ fontSize: "20px" }}
			>
				<button>Modal</button>
			</Modal>
		);
		expect(getByTestId("test")).toMatchSnapshot();
	});

	describe("theming", () => {
		const themeTest: ThemeConfig = {
			mode: "mode",
			Modal: {
				styles: {
					root: {
						mt: 10,
						letterSpacing: 250,
					},
					modal: {
						mt: 11,
						letterSpacing: 251,
					},
					inner: {
						mt: 12,
						letterSpacing: 252,
					},
					container: {
						mt: 13,
						letterSpacing: 254,
					},
					overlay: {
						mt: 14,
						letterSpacing: 255,
					},
				},
				variants: {
					custom: {
						root: {
							top: 0,
							textIndent: 1,
						},
						modal: {
							top: 1,
							textIndent: 2,
						},
						overlay: {
							top: 2,
							textIndent: 3,
						},
						container: {
							top: 3,
							textIndent: 4,
						},
						inner: {
							top: 4,
							textIndent: 5,
						},
					},
				},
			},
		};

		it("should extend styles in global", () => {
			const { getByTestId } = render(
				<Modal data-testid="test" isOpen={true} variant="custom">
					<button>Modal</button>
				</Modal>,
				{
					theme: themeTest,
				}
			);
			expect(getByTestId("test")).toMatchSnapshot();
		});

		it("should extend styles in local", () => {
			const { getByTestId } = render(
				<Modal
					data-testid="test"
					isOpen={true}
					variant="custom"
					themeExtend={themeTest}
				>
					<button>Modal</button>
				</Modal>
			);
			expect(getByTestId("test")).toMatchSnapshot();
		});

		it("should render with default props correctly", () => {
			const { getByTestId } = render(
				<Modal data-testid="test" isOpen={true}>
					<button>Modal</button>
				</Modal>,
				{
					theme: {
						Modal: {
							defaultProps: {
								id: "id",
								className: "hello",
							},
						},
					},
				}
			);
			expect(getByTestId("test")).toMatchSnapshot();
		});

		it("should override styles", () => {
			const { getByTestId } = render(
				<Modal data-testid="test" isOpen={true}>
					<button>Modal</button>
				</Modal>,
				{
					theme: {
						Modal: {
							overrides: {
								root: {
									mt: 50,
								},
							},
						},
					},
				}
			);
			expect(getByTestId("test")).toMatchSnapshot();
		});
	});
});
