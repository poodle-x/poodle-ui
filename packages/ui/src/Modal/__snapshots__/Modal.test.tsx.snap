// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> ref should work 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 2000;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
}

.c10 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
}

.c13 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
}

<div
  data-portal="poodle-ui"
>
  <div
    class="poodle-box c0 poodle-modal c1 c2"
  >
    <div
      aria-hidden="true"
      class="poodle-box c0 poodle-modal__overlay c4 c2"
    />
    <div
      class="poodle-box c0 poodle-modal__inner c7 c2"
    >
      <div
        class="poodle-box c0 poodle-modal__container c10 c2"
        tabindex="-1"
      >
        <div
          aria-modal="true"
          class="poodle-box c0 poodle-modal__modal c13 c2"
          role="dialog"
        >
          <button>
            Modal
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> should render Box props correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 2000;
}

.c2 {
  margin-top: 20px;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
}

.c5 {
  margin-left: 20px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
}

.c8 {
  padding-left: 20px;
}

.c10 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
}

.c13 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
}

.c14 {
  font-size: 20px;
}

<div
  class="poodle-box c0 poodle-modal c1 c2"
  data-testid="test"
  id="id"
>
  <div
    aria-hidden="true"
    class="poodle-box c0 poodle-modal__overlay c4 c5"
  />
  <div
    class="poodle-box c0 poodle-modal__inner c7 c8"
  >
    <div
      class="poodle-box c0 poodle-modal__container c10 c8"
      tabindex="-1"
    >
      <div
        aria-modal="true"
        class="poodle-box c0 poodle-modal__modal c13 c14"
        role="dialog"
      >
        <button>
          Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> should render Modal correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 2000;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
}

.c10 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
}

.c13 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
}

<div
  class="poodle-box c0 poodle-modal c1 c2"
  data-testid="test"
>
  <div
    aria-hidden="true"
    class="poodle-box c0 poodle-modal__overlay c4 c2"
  />
  <div
    class="poodle-box c0 poodle-modal__inner c7 c2"
  >
    <div
      class="poodle-box c0 poodle-modal__container c10 c2"
      tabindex="-1"
    >
      <div
        aria-modal="true"
        class="poodle-box c0 poodle-modal__modal c13 c2"
        role="dialog"
      >
        <button>
          Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> should render Modal withAutoId correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 2000;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
}

.c10 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
}

.c13 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
}

<div
  class="poodle-box c0 poodle-modal c1 c2"
  data-testid="test"
>
  <div
    aria-hidden="true"
    class="poodle-box c0 poodle-modal__overlay c4 c2"
  />
  <div
    class="poodle-box c0 poodle-modal__inner c7 c2"
  >
    <div
      class="poodle-box c0 poodle-modal__container c10 c2"
      tabindex="-1"
    >
      <div
        aria-describedby="poodle-modal-describe-2"
        aria-labelledby="poodle-modal-label-1"
        aria-modal="true"
        class="poodle-box c0 poodle-modal__modal c13 c2"
        role="dialog"
      >
        <button>
          Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> theming should disable classes 1`] = `
.c0 {
  box-sizing: border-box;
}

.c3 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
}

.c9 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
}

.c12 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
}

<div
  class="poodle-box c0 poodle-modal c1"
  data-testid="test"
>
  <div
    aria-hidden="true"
    class="poodle-box c0 poodle-modal__overlay c3 c1"
  />
  <div
    class="poodle-box c0 poodle-modal__inner c6 c1"
  >
    <div
      class="poodle-box c0 poodle-modal__container c9 c1"
      tabindex="-1"
    >
      <div
        aria-modal="true"
        class="poodle-box c0 poodle-modal__modal c12 c1"
        role="dialog"
      >
        <button>
          Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> theming should extend styles in global 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 2000;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
  margin-top: 14px;
  letter-spacing: 255px;
  top: 2px;
  text-indent: 3px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
  margin-top: 12px;
  letter-spacing: 252px;
  top: 4px;
  text-indent: 5px;
}

.c10 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
  margin-top: 13px;
  letter-spacing: 254px;
  top: 3px;
  text-indent: 4px;
}

.c13 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
  margin-top: 11px;
  letter-spacing: 251px;
  top: 1px;
  text-indent: 2px;
}

<div
  class="poodle-box c0 poodle-modal c1 c2"
  data-testid="test"
>
  <div
    aria-hidden="true"
    class="poodle-box c0 poodle-modal__overlay c4 c2"
  />
  <div
    class="poodle-box c0 poodle-modal__inner c7 c2"
  >
    <div
      class="poodle-box c0 poodle-modal__container c10 c2"
      tabindex="-1"
    >
      <div
        aria-modal="true"
        class="poodle-box c0 poodle-modal__modal c13 c2"
        role="dialog"
      >
        <button>
          Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> theming should extend styles in local 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 2000;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
  margin-top: 14px;
  letter-spacing: 255px;
  top: 2px;
  text-indent: 3px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
  margin-top: 12px;
  letter-spacing: 252px;
  top: 4px;
  text-indent: 5px;
}

.c10 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
  margin-top: 13px;
  letter-spacing: 254px;
  top: 3px;
  text-indent: 4px;
}

.c13 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
  margin-top: 11px;
  letter-spacing: 251px;
  top: 1px;
  text-indent: 2px;
}

<div
  class="poodle-box c0 poodle-modal c1 c2"
  data-testid="test"
>
  <div
    aria-hidden="true"
    class="poodle-box c0 poodle-modal__overlay c4 c2"
  />
  <div
    class="poodle-box c0 poodle-modal__inner c7 c2"
  >
    <div
      class="poodle-box c0 poodle-modal__container c10 c2"
      tabindex="-1"
    >
      <div
        aria-modal="true"
        class="poodle-box c0 poodle-modal__modal c13 c2"
        role="dialog"
      >
        <button>
          Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> theming should override classes 1`] = `
.c0 {
  box-sizing: border-box;
}

.c3 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
}

.c9 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
}

.c12 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
}

<div
  class="poodle-box c0 poodle-modal morning c1"
  data-testid="test"
>
  <div
    aria-hidden="true"
    class="poodle-box c0 poodle-modal__overlay c3 c1"
  />
  <div
    class="poodle-box c0 poodle-modal__inner c6 c1"
  >
    <div
      class="poodle-box c0 poodle-modal__container c9 c1"
      tabindex="-1"
    >
      <div
        aria-modal="true"
        class="poodle-box c0 poodle-modal__modal c12 c1"
        role="dialog"
      >
        <button>
          Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> theming should override styles 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  margin-top: 50px;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
}

.c10 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
}

.c13 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
}

<div
  class="poodle-box c0 poodle-modal c1 c2"
  data-testid="test"
>
  <div
    aria-hidden="true"
    class="poodle-box c0 poodle-modal__overlay c4 c2"
  />
  <div
    class="poodle-box c0 poodle-modal__inner c7 c2"
  >
    <div
      class="poodle-box c0 poodle-modal__container c10 c2"
      tabindex="-1"
    >
      <div
        aria-modal="true"
        class="poodle-box c0 poodle-modal__modal c13 c2"
        role="dialog"
      >
        <button>
          Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> theming should render with default props correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 2000;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: rgba(0, 0, 0, 0.25);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  outline: none;
}

.c10 {
  max-height: calc(100% - 80px);
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 40px;
  margin-bottom: 40px;
  width: 100%;
  max-width: 600px;
}

.c13 {
  width: 100%;
  max-height: 100%;
  border-radius: 4px;
  background-color: #F8F8F8;
  box-shadow: 1px 2px 2px -1px rgba(0,0,0,0.4);
}

<div
  class="poodle-box c0 poodle-modal c1 hello c2"
  data-testid="test"
  id="id"
>
  <div
    aria-hidden="true"
    class="poodle-box c0 poodle-modal__overlay c4 c2"
  />
  <div
    class="poodle-box c0 poodle-modal__inner c7 c2"
  >
    <div
      class="poodle-box c0 poodle-modal__container c10 c2"
      tabindex="-1"
    >
      <div
        aria-modal="true"
        class="poodle-box c0 poodle-modal__modal c13 c2"
        role="dialog"
      >
        <button>
          Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;
